overlord/overlord.go

```diff
@@ -175,11 +176,13 @@ func New(restartHandler restart.Handler) (*Overlord, error) {
 	}
 	o.addManager(ifaceMgr)
 
-	fdeMgr, err := fdestate.Manager(s, o.runner)
-	if err != nil {
-		return nil, err
+	if secboot.WithSecbootSupport {
+		fdeMgr, err := fdestate.Manager(s, o.runner)
```

Comment:
hm a nil manager may have a quirky side effect if not all code accounts for it and I would prefer to not start adding secboot support check at random places. Would it make sense to store an initialization error like we already do, we can store the error in fdestate.FDEManager.initErr and only surface it in the API calls? Perhaps even something like this in the fdestate backend?

```
func IsSupported() bool {
    return secboot.WithSecbootSupport
}
```

Ideally in the long run, I think we should inject the backend into the manager like we do in snapstate